### PR TITLE
DEP PHP Support in CMS5

### PR DESCRIPTION
**Pull request by sabina-talipova**
## Parent issue
- https://github.com/silverstripe/silverstripe-framework/issues/10402

### CODE REVIEW

**Comment by sabina-talipova** [comment]
Requires https://github.com/silverstripe/silverstripe-widgets/pull/193 and https://github.com/silverstripe/silverstripe-comments/pull/354

**Comment by sabina-talipova** [comment]
Created new card for failed Behat tests https://github.com/silverstripe/silverstripe-blog/issues/696

File: tests/php/BlogFunctionalTest.php [hunk 1]
```diff
@@ -11,8 +11,6 @@ class BlogFunctionalTest extends FunctionalTest
 {
     protected static $fixture_file = 'BlogFunctionalTest.yml';
 
-    protected static $use_draft_site = true;
```
Comment (maxime-rainville) [inline tests/php/BlogFunctionalTest.php:14]: Why did we change this?

File: tests/php/BlogFunctionalTest.php [hunk 2]
```diff
@@ -23,22 +21,25 @@ protected function setUp(): void
 
     public function testBlogWithMultibyteUrl()
     {
-        $result = $this->get(rawurlencode('آبید'));
+        $this->logInWithPermission('ADMIN');
```
Comment (maxime-rainville) [inline tests/php/BlogFunctionalTest.php]: Why was this changed?

**Review by maxime-rainville — COMMENTED** [review]
Looks fine. Just curious about why the changes to the test were needed.

File: tests/php/BlogFunctionalTest.php (hunk 1 above)
Comment (maxime-rainville) [reply tests/php/BlogFunctionalTest.php:14]: `use_draft_site` has been deprecated. 

https://github.com/silverstripe/silverstripe-framework/blob/2c105cffc9436a0587381956118fe3361bad04d6/src/Dev/FunctionalTest.php#L70

File: tests/php/BlogFunctionalTest.php (hunk 2 above)
Comment (maxime-rainville) [reply tests/php/BlogFunctionalTest.php]:
```suggestion
        $this->logInWithPermission('VIEW_DRAFT_CONTENT');
```

File: tests/php/BlogFunctionalTest.php [hunk 3]
```diff
@@ -23,22 +21,25 @@ protected function setUp(): void
...
+        $result = $this->get(rawurlencode('آبید') . '?stage=Stage');
 
         $this->assertEquals(200, $result->getStatusCode());
     }
 
     public function testMemberProfileWithMultibyteUrlAndName()
     {
-        $result = $this->get(rawurlencode('آبید') . '/profile/' . rawurlencode('عبّاس-آبان'));
+        $this->logInWithPermission('ADMIN');
```
Comment (maxime-rainville) [inline tests/php/BlogFunctionalTest.php]:
```suggestion
        $this->logInWithPermission('VIEW_DRAFT_CONTENT');
```

File: tests/php/BlogFunctionalTest.php [hunk 4]
```diff
@@ -23,22 +21,25 @@ protected function setUp(): void
...
+        $result = $this->get(rawurlencode('آبید') . '/profile/' . rawurlencode('عبّاس-آبان') . '?stage=Stage');
 
         $this->assertEquals(200, $result->getStatusCode());
         $this->assertStringContainsString('My Blog Post', $result->getBody());
     }
 
     public function testMemberProfileWithMultibyteUrlAndEnglishName()
     {
-        $result = $this->get(rawurlencode('آبید') . '/profile/bob-jones');
+        $this->logInWithPermission('ADMIN');
```
Comment (maxime-rainville) [inline tests/php/BlogFunctionalTest.php]:
```suggestion
        $this->logInWithPermission('VIEW_DRAFT_CONTENT');
```

File: tests/php/Model/BlogControllerFunctionalTest.php [hunk 5]
```diff
@@ -23,15 +21,17 @@ protected function setUp(): void
 
     public function testGetCategoriesWithMultibyteUrl()
     {
-        $result = $this->get('my-blog/category/' . rawurlencode('آبید'));
+        $this->logInWithPermission('ADMIN');
```
Comment (maxime-rainville) [inline tests/php/Model/BlogControllerFunctionalTest.php]:
```suggestion
        $this->logInWithPermission('VIEW_DRAFT_CONTENT');
```

File: tests/php/Model/BlogControllerFunctionalTest.php [hunk 6]
```diff
@@ -23,15 +21,17 @@ protected function setUp(): void
...
+        $result = $this->get('my-blog/category/' . rawurlencode('آبید') . '?stage=Stage');
 
         $this->assertEquals(200, $result->getStatusCode());
         $this->assertStringContainsString('آبید', $result->getBody());
     }
 
     public function testGetTagsWithMultibyteUrl()
     {
-        $result = $this->get('my-blog/tag/' . rawurlencode('برتراند'));
+        $this->logInWithPermission('ADMIN');
```
Comment (maxime-rainville) [inline tests/php/Model/BlogControllerFunctionalTest.php]:
```suggestion
        $this->logInWithPermission('VIEW_DRAFT_CONTENT');
```

File: tests/php/BlogFunctionalTest.php (hunk 1 above)
Comment (sabina-talipova) [reply tests/php/BlogFunctionalTest.php:14]: $use_draft_site is deprecated. We removed it [here](https://github.com/silverstripe/silverstripe-framework/pull/10594/files#diff-542dfb607458124757d02f52f7a504dab59b280168ee8574a89e163c3f64581aL46-L53)

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Answered my own question.

I just think we should use the more minimal `VIEW_DRAFT_CONTENT` permission to run the test.

File: tests/php/BlogFunctionalTest.php (hunk 2 above)
Comment (sabina-talipova) [reply tests/php/BlogFunctionalTest.php]: User doesn't have an access to the Draft page without Admin permission

Comment (sabina-talipova) [reply tests/php/BlogFunctionalTest.php]: DONE

File: tests/php/BlogFunctionalTest.php (hunk 3 above)
Comment (sabina-talipova) [reply tests/php/BlogFunctionalTest.php]: DONE

File: tests/php/BlogFunctionalTest.php (hunk 4 above)
Comment (sabina-talipova) [reply tests/php/BlogFunctionalTest.php]: DONE

File: tests/php/Model/BlogControllerFunctionalTest.php (hunk 5 above)
Comment (sabina-talipova) [reply tests/php/Model/BlogControllerFunctionalTest.php]: DONE

File: tests/php/Model/BlogControllerFunctionalTest.php (hunk 6 above)
Comment (sabina-talipova) [reply tests/php/Model/BlogControllerFunctionalTest.php]: DONE

**Review by maxime-rainville — APPROVED** [review]
LGTM

**Pull request merged**